### PR TITLE
Add inline stylesheets

### PR DESCRIPTION
**Pull request by simonewebdesign**

### CODE REVIEW

File: src/inlineStylesheets.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env node
+
+const fs = require('fs').promises;
+const path = require('path');
+
+let inlineHtmlStyles = async htmlPath => {
+	const linkTagRegex = /<link.+href="([\w.\-\/]+)css".*>/;
```
Comment (mahhov) [inline src/inlineStylesheets.js]: Minor points:
1. Let's enforce the whitespace after 'link'
2. Not enforce a '.css' filename suffix. I'd rather check for ' rel="stylesheet" ' .

```suggestion
	const linkTagRegex = /<link (?:.* )?rel="stylesheet"(?:.* )?href="([\w.\-\/]+)".*>|<link (?:.* )?href="([\w.\-\/]+)"(?:.* )?rel="stylesheet".*>/;
```

3. for consistency, could you likewise add `\-` to the regexes in inlineRequires.js and inlineScriptTags.js?
Thanks for catching this case.

File: src/inlineStylesheets.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	let html = await fs.readFile(htmlPath, 'utf8');
+	let stylesheetPromises = html
+		.match(new RegExp(linkTagRegex, 'g'))
+		.map(linkTag => linkTag.match(linkTagRegex)[1])
+		.map(relPath => path.resolve(path.dirname(htmlPath), relPath + 'css'))
```
Comment (mahhov) [inline src/inlineStylesheets.js]:
```suggestion
		.map(relPath => path.resolve(path.dirname(htmlPath), relPath))
```

File: readme.md [hunk 3]
```diff
@@ -28,13 +30,36 @@ console.log('Welcome');
...
+	color: red;
+}
+```
+
+`$ inline-stylesheets src/index.html out/index.html`
+
+```html
+<!-- out/index.html -->
+<p>Welcome</p>
+<style>body { color: red; }</style>
```
Comment (mahhov) [inline readme.md]:
```suggestion
<style>body { 
	color: red; 
}</style>
```
Minor: just whitespace accuracy with what the script actually produces

File: src/inlineStylesheets.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+const fs = require('fs').promises;
+const path = require('path');
+
+let inlineHtmlStyles = async htmlPath => {
+	const linkTagRegex = /<link.+href="([\w.\-\/]+)css".*>/;
+	let html = await fs.readFile(htmlPath, 'utf8');
+	let stylesheetPromises = html
+		.match(new RegExp(linkTagRegex, 'g'))
+		.map(linkTag => linkTag.match(linkTagRegex)[1])
```
Comment (mahhov) [inline src/inlineStylesheets.js]:
```suggestion
		.map(linkTag => {
			let m = linkTag.match(linkTagRegex);
			return m[1] || m[2];
		})
```

File: test/testInlineStylesheets/main.css [hunk 5]
```diff
@@ -0,0 +1,3 @@
+body {
```
Comment (mahhov) [inline test/testInlineStylesheets/main.css]: I don't think this file is used? If not, please remove. 
And if you want to, also remove test/testInlineScriptTags/main.js (also unused. I think this was a copy paste leftover).

File: test/testInlineStylesheets/index.html [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<link href="/author.html" rel="author">
```
Comment (mahhov) [inline test/testInlineStylesheets/index.html:4]:
```suggestion
<link href="/author.html" rel="author">
<link href="myStyle.css">
<linkrel="stylesheet" href="myStyle.css">
```

File: test/testInlineStylesheets/index.html [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+<p>line 1</p>
+
+<link rel="stylesheet" href="myStyle.css">
```
Comment (mahhov) [inline test/testInlineStylesheets/index.html]:
```suggestion
<link rel="stylesheet" href="myStyle.css">
<link href="myStyle.css"rel="stylesheet">
```

**Review by mahhov — CHANGES_REQUESTED** [review]
Thank you for the pull request!

I have some minor comments.

Also, please revert the 3 files with no changes (src/cli/inlineScriptTags.js, src/cli/inlineRequires.js, src/cli/inlineEnvironmentVariables.js)

File: test/testInlineStylesheets/main.css (hunk 5 above)
Comment (simonewebdesign) [reply test/testInlineStylesheets/main.css]: I removed main.css and there were two html files for inlineScriptTags, so I removed myHtml.html and only left one index.html (see d476ffe)

**Comment by simonewebdesign** [comment]
I think I've addressed all the comments, could you please double check? Thanks :)

File: test/testInlineStylesheets/index.html [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<link href="/author.html" rel="author">
+<link href="myStyle.css" rel="stylesheet">
+<link href="nested/myNestedStylesheet.css" rel="stylesheet" />
```
Comment (mahhov) [inline test/testInlineStylesheets/index.html]: This is good for checking the replacement works. But I also want to make sure the non-trivial regex is working.

```
Link tags without `rel="stylesheet"` should not be replaced.

Not a stylesheet
<link href="/author.html" rel="author">

Missing rel attribute
<link href="myStyle.css">

Incorrect syntax
<linkrel="stylesheet" href="myStyle.css">


Link tags with `rel="stylesheet"` should be replaced.

<link href="myStyle.css" rel="stylesheet">

Space between attributes is optional
<link rel="stylesheet"href="myStyle.css">

<link href="nested/myNestedStylesheet.css" rel="stylesheet" />
```

**Review by mahhov — CHANGES_REQUESTED** [review]
Awesome, just 1 more comment regarding the stylesheet tests.

File: test/testInlineStylesheets/index.html (hunk 8 above)
Comment (simonewebdesign) [reply test/testInlineStylesheets/index.html]: Ah, I see now. I've updated the test, seems to work fine!

**Comment by mahhov** [comment]
Thanks!

**Pull request merged**